### PR TITLE
chore(weave): Fix table refs rendering

### PR DESCRIPTION
**Pull request by tssweeney**
Currently (both before and after the calls table refactor), when expanding "Dataset" objects (or any object with a "table ref"), the resulting cells for such fields result in the mini cell error boundary (gray bubble with error).

This was occurring due to some compounding factors. However, the key insight is:
1. "table refs" (those that are of the form `weave://ent/proj/table/version`) are special and not intended to be used directly
2. When expanding objects that contain table refs - these refs come back from the server
3. Our current processing was treating them like normal refs (eg. expansion analysis & SmallRef rendering/linking)
4. Table refs never make sense in isolation - they always need to be paired with their source ref.

Ok, with all that considered, the main change here is to post-process the calls data and transform table refs into a special object containing the source ref and the table ref such that the UI can appropriately render/process this data.

Much of the LoC changes here are moving symbols around

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/ObjectViewer.tsx [hunk 1]
```diff
@@ -59,23 +59,6 @@ const getRefs = (data: Data): string[] => {
 
 type RefValues = Record<string, any>; // ref URI to value
 
-const refIsExpandable = (ref: string): boolean => {
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/ObjectViewer.tsx:62]: This was moved to `weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksCallRefExpansion.ts`

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTable.tsx [hunk 2]
```diff
@@ -638,75 +631,3 @@ const ExportRunsTableButton = ({
     </Button>
   </Box>
 );
-
-/**
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTable.tsx:642]: moved to `weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableDataProcessing.ts`

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableDataProcessing.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+ * be consumed by the CallsTable. This format contains both the source ref (the
+ * path from the object ref to the table data) as well as the actual table ref
+ * itself. This allows the calls table to appropriately render a ref link to the
+ * "source". We cannot just replace the ref with the source ref however, since
+ * the calls table doesn't have any way to "know" that it is a table ref. As a
+ * result, it would think that this ref is expandable, which is not a valid
+ * operation. Therefore, we use this special type
+ * `ExpandedRefWithValueAsTableRef` to handle this case.
+ */
+function replaceTableRefsInFlattenedData(flattened: Record<string, any>) {
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableDataProcessing.ts:106]: This is the main "functional change" of this PR

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksCallRefExpansion.ts [hunk 4]
```diff
@@ -174,3 +175,37 @@ export const directFetchRefsData = async (
 
   return refUris.map(uri => cached[uri]);
 };
+
+export const isTableRef = (ref: any): boolean => {
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksCallRefExpansion.ts:179]: just some conveniences.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableDataProcessing.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+      // and add the cleaned key to the cleaned object
+      cleaned[newKey] = flattened[key];
+    });
+
+    return cleaned as TraceCallSchema & {[key: string]: string};
+  });
+}
+/**
+ * This processing step is admittedly a bit of a hack. It replaces "table refs"
+ * (those that are of weaveKind "table" with a special format that can be
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableDataProcessing.ts:86]: close paren missing?

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableDataProcessing.ts [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+ * consumed by the CallsTable. THe reason we don't want to leave plain table
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableDataProcessing.ts:87]: THe => The

**Review by jamie-rasmussen — APPROVED** [review]

**Pull request merged**